# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [5]:
print(data.groupby('income_type')['total_income'].count())

income_type
безработный            2
в декрете              1
госслужащий         1312
компаньон           4577
пенсионер           3443
предприниматель        1
сотрудник          10014
студент                1
Name: total_income, dtype: int64


Выведем на экран количество заполненных ячеек дохода по группам занятости

In [6]:
print(data.groupby('income_type')['days_employed'].count()) 

income_type
безработный            2
в декрете              1
госслужащий         1312
компаньон           4577
пенсионер           3443
предприниматель        1
сотрудник          10014
студент                1
Name: days_employed, dtype: int64


Выведем на экран количество заполненных ячеек столбца "стаж" по группам занятости. Количество данных в ячейках одинаковое - значит пропуски расположены в одних и тех же строках.

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [7]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [8]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [9]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [10]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [11]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [12]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [13]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [14]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [15]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [16]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [17]:
data.duplicated().sum()

71

In [18]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [19]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [20]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [21]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [22]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [23]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [24]:
data_pivot_children = data.pivot_table(index=['children'], values='debt', aggfunc=['sum','count']) # Ваш код будет здесь. Вы можете создавать новые ячейки.
# с помощью pivot_table создадим сводную таблицу из представленного датафрейма

In [25]:
print(data_pivot_children) # выведем сводную таблицу на экран

           sum  count
          debt   debt
children             
0         1063  14091
1          444   4808
2          194   2052
3           27    330
4            4     41
5            0      9


Получившаяся таблица не позволяет наглядно сделать вывод о зависимости между количеством детей и возвратом кредита в срок, так как содержит только количество просроченных платежей(sum debt) и общее количество(count debt). Для придания сводной таблице наглядности, добавим новый столбец, рассчитывающий долю просроченных платежей от общего количества платежей.

In [26]:
def ratio(pdSerises):
    return str(round((pdSerises.sum() / pdSerises.count()) * 100, 2)) + '%'
# создаем собственную функцию

In [27]:
data_pivot_children = data.pivot_table(index=['children'], values='debt', aggfunc=['sum','count',ratio])
# с помощью pivot_table создаем сводную таблицу уже с учетом процентного соотношения

In [28]:
print(data_pivot_children)
# выведем сводную таблицу на экран

           sum  count  ratio
          debt   debt   debt
children                    
0         1063  14091  7.54%
1          444   4808  9.23%
2          194   2052  9.45%
3           27    330  8.18%
4            4     41  9.76%
5            0      9   0.0%


Сводная таблица становится более читаемой и пригодной для анализа зависимости между количеством детей и возвратом кредита в срок. Можно отметить, что многодетные семьи (семьи, где 3 и более ребенка) неохотно берут кредиты (невысокое количество взятых кредитов). Таким образом, из-за того, что количество замещиков, у которых больше двух детей, сильно отличается по количеству от остальной выборки, есть смысл объединить их в одну категорию. 

In [29]:
# Поскольку больше двух детей данных в принципе не очень много объединяем их в одну категорию
def make_children_category(children):
    if children == 1:
        return '1 ребенок'
    if children == 2:
        return '2 ребенка'
    if children >= 3:
        return 'Много детей'
    return 'Нет детей'

data['children_category'] = data['children'].apply(make_children_category)

In [30]:
# проверим, что все получилось
print(data['children_category'].value_counts())

Нет детей      14091
1 ребенок       4808
2 ребенка       2052
Много детей      380
Name: children_category, dtype: int64


Повторяем свои действия по получению сводной таблицы

In [31]:
data_pivot_children = data.pivot_table(index=['children_category'], values='debt', aggfunc=['sum','count',ratio])

In [32]:
print(data_pivot_children)

                    sum  count  ratio
                   debt   debt   debt
children_category                    
1 ребенок           444   4808  9.23%
2 ребенка           194   2052  9.45%
Много детей          31    380  8.16%
Нет детей          1063  14091  7.54%


**Вывод:** Наибольший процент невозврата кредита в срок у заемщиков с 2 детьми. Самые надежные заемщики с самой низкой долей просрочек по кредиту - это заемщики без детей. Таким образом, при появлении каждого нового ребенка увеличивается кредитная нагрузка, что влияет на способность возврата кредита. Кредит не бояться брать те, у кого нет детей, а чем детей больше - тем меньше желающих.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [33]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
data_pivot_family_status = data.pivot_table(index=['family_status'], values='debt', aggfunc=['sum','count',ratio])

In [34]:
print(data_pivot_family_status)

                       sum  count  ratio
                      debt   debt   debt
family_status                           
Не женат / не замужем  273   2796  9.76%
в разводе               84   1189  7.06%
вдовец / вдова          63    951  6.62%
гражданский брак       385   4134  9.31%
женат / замужем        927  12261  7.56%


**Вывод:** Да, зависимость есть. Наибольший процент невозврата кредита в срок у одиноких заемщиков (не женат/не замужем – 9,76% ). Но, те кто в разводе (7,06%) или со статусом вдовец/вдова (6,62%) чаще платят в срок, чем люди в браке (7,56%). 

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [35]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# Уровень дохода тоже нужно определить по категориям и присовить им наименования, удобные для анализа.
# Я создала функцию и выделила следующие категории дохода. Они соответствуют ранее проведенному анализу по категориям.
def make_name_total_income(total_income):
    if total_income <= 30000:
        return 'до 30'
    if total_income <= 50000:
        return 'от 30 до 50'
    if total_income <= 200000:
        return 'от 50 до 200'
    if total_income <= 1000000:
        return 'от 200 до 1000'
    return 'больше 1000'

data['name_total_income'] = data['total_income'].apply(make_name_total_income)

In [36]:
# проверим результат
print(data['name_total_income'].value_counts())

от 50 до 200      15921
от 200 до 1000     5014
от 30 до 50         349
больше 1000          25
до 30                22
Name: name_total_income, dtype: int64


In [37]:
data_pivot_total_income = data.pivot_table(index=['name_total_income'], values='debt', aggfunc=['sum','count',ratio])  

In [38]:
print(data_pivot_total_income)

                    sum  count  ratio
                   debt   debt   debt
name_total_income                    
больше 1000           2     25   8.0%
до 30                 2     22  9.09%
от 200 до 1000      354   5014  7.06%
от 30 до 50          21    349  6.02%
от 50 до 200       1353  15921   8.5%


**Вывод:** Самая малочисленная группа по взятию кредита - это потенциальные заемщики с уровнем дохода менее 30 тыс рублей, в этой же группе самая высокая доля просрочек по кредиту: доля невозврата в срок 9,09%. Также малочисленная группа по взятию кредита, это люди с доходом более 1 млн рублей. Таким образом, судя по уровню дохода, кредит не особо интересен тем, кто зарабатывает очень много, как и тем, кто зарабатывает мало. Основная группа заемщиков - люди с доходом от 50 до 200 тыс рублей. При росте дохода (при переходе  из группы "от 50 до 200" в группу "от 200 до 1000") уменьшается вероятность невозврата кредита в срок. Делаю вывод, что исправнее всего платят заемщики с доходом от 200 до 1000 тыс. рублей.
Получается, самая "опасная" группа потенциальных заемщиков - это те заемщики, у которых доход от 50 до 200 тыс.рублей, так называемый условный средний класс: количество таких заемщиков больше всего среди остальных групп, и процент невозврата кредита в срок не самый маленький.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [39]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
data_pivot_purpose_category = data.pivot_table(index=['purpose_category'], values='debt', aggfunc=['sum','count',ratio])

In [40]:
print(data_pivot_purpose_category)

                          sum  count  ratio
                         debt   debt   debt
purpose_category                           
операции с автомобилем    400   4279  9.35%
операции с недвижимостью  780  10751  7.26%
получение образования     369   3988  9.25%
проведение свадьбы        183   2313  7.91%


**Вывод:** Операции с недвижимостью - самая популярная категория среди кредитов, и в этой же категории самые лучшие показатели возврата в срок. Для этого могут быть следующие причины: люди, покупающие жилье, планируют семейную жизнь, или у них уже есть семья. Это чаще всего люди с постоянным доходом и осмысленным принятием такого важного решения, как покупка недвижимости.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Основные пропуски в исходных данных наблюдались в следующих столбцах:days_imployed(общий трудовой стаж в днях) и total_income (ежемесячный доход). Причиной пропуска значений в столбце days_imployed(общий трудовой стаж в днях) может быть:
1. ошибка при заведении информации в базу, неправильный ввод данных.
2. отсутствие возможности получения или обработки данных.

Причиной пропуска значений в столбце total_income (ежемесячный доход) может быть:
1. ошибка при заведении информации в базу, неправильный ввод данных.
2. нежелание разглашать информацию о доходах. 

Основные пропуски в исходных данных наблюдались в следующих столбцах: days_imployed (общий трудовой стаж в днях) и total_income (ежемесячный доход). Выше, я дополнительно сделала расчет количества пропусков, а также где они располагаются. Количество пропусков в двух вышеназванных столбцах составило по 2174 шт., то есть равное количество в двух столбцах. Построчное сравнение выявило тот факт, что значения отсутствуют в одних и тех же строках. То есть, люди не указывали стаж и доход одновременно. Так как наименования типов занятости носят размытый характер (под «сотрудником» может подразумеваться, как продавец в магазине, так и курьер, и сотрудник банка и пр.) пропуски в доходах и стаже наблюдаются среди всех многочисленных представленных групп. Таким образом, невозможно сделать о вывод о наличии пропусков только у одной группы занятых.

Возможной причиной пропуска данных в столбце ежемесячный доход может являться тот факт, что потенциальный заемщик является «зарплатным» клиентом, то есть имеет карту банка и получает на нее заработную плату, таким образом, у банка есть информация о доходах заемщика, а также о его занятости (ведь это "зарплатная" карта). Эта же самая причина может относиться к пенсионерам, которые получают пенсию на карту банка

По возможной причине пропуска данных в столбце общий трудовой стаж: в некоторых случаях Трудовой кодекс предполагает работу по трудовому договору, но без занесения сведений об этом в трудовую книжку. Таким образом, стаж работы по контракту (договору ГПХ) нигде учитывается: ни при оформлении пособия по безработице, ни при оформлении больничного листа, контракт также не предполагает оплачиваемый отпуск. Сделаю предположение, что и банк может не рассматривать такой стаж в связи с отсутствием подтвержденной записи в трудовой книжке. 

Действительно, в наших данных пропуски в столбцах "трудовой стаж" и "ежемесячный доход" абсолютно идентичны. Количество строк с отсутствующими данными в обоих указанных столбцах совпадает и равно 2174. Маловероятно, что специалисты по выдаче кредитов, заполнив все без исключения ячейки других столбцов по каждому кредитополучателю, оставляли незаполненными именно два показателя, и сделали это ровно 2174 раза в каждом столбце. Даже не проводя проверку, во всех ли случаях незаполненные ячейки находятся в одной и то же строке с высокой долей вероятности можно утверждать, что произошла техническая ошибка при формировании таблицы с исходными данными (либо некий сбой во время выгрузки данных на анализ, либо при объединении данных в единый файл, который также должен был передаваться на анализ, либо эти данные не собирались). Это моё ИМХО (врачи говорят – «второе мнение»).

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Медиана — это число в середине упорядоченного набора чисел (выборки): половина данных находится ниже этого значения, а половина выше. Согласно определению, среднее арифметическое зависит от всех элементов выборки, наличие экстремальных значений значительно влияет на результат. В таких ситуациях, как наша выборка, где представлены доходы и до 30 тыс рублей и свыше 1 млн рублей, и количество данных в этих двух группах невелико, среднее арифметическое может исказить смысл числовых данных и привести к неверному выводу. Поэтому замена пропусков медианным значением  - это лучшее решение.

### Шаг 4: общий вывод.

Напишите ваш общий вывод. В результате проделанной работы по анализу зависимости семейного положения и количества детей клиента на факт погашения кредита в срок можно сделать вывод, что идеальными заемщиками для банка при рассмотрении заявок на кредит являются:
1. желающие взять кредит для покупки недвижимости (ипотеку).
2. семейные люди без детей.
3. с уровнем дохода от 200 тыс.рублей до 1 млн. рублей

 Новый общий вывод:
Было проведено исследование, в ходе которого надо было разобраться, как влияют различные параметры (семейное положение, количество детей, доход клиента, а также цели кредита) на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов. Результаты исследования будут учтены при построении модели кредитного скоринга – специальной системы, которая оценивает способность потенциального заемщика вернуть кредит банку. Считается, что люди с похожими параметрами примерно одинаково обращаются с финансами. Поэтому банки сравнивают потенциальных заемщиков с другими людьми, которые уже брали похожие кредиты.

В ходе исследования были выдвинуты следующие предположения для проверки:

    Предположение 1: количество детей влияет на погашение кредита в срок
    Предположение 2: семейное положение влияет на факт погашения кредита в срок
    Предположение 3: уровень дохода влияет на факт погашения кредита в срок
    Предположение 4: цель кредита влияет на факт погашения кредита в срок.

В результате удалось выявить следующие зависимости.

Предположение 1 частично подтверждено: для более точных выводов необходимо больше данных по клиентам, у которых 3 и более ребенка.

    7,54% - доля просроченных кредитов у заемщиков без детей
    9,23% - доля просроченных кредитов у заемщиков с 1 ребенком
    9,45% - доля просроченных кредитов у заемщиков с 2 детьми

Рекомендация: провести дополнительное исследование на большей выборке данных.

Предположение 2 подтверждено:

    у неженатых клиентов(9,76%), и у тех, кто состоит в гражданском браке (9,31%) выше вероятность стать должником.
    те, кто в разводе (7,06%), женат/замужем(7,56%) или вдовец/вдова (6,62%) менее вероятно становятся должниками.

Рекомендация: разница между высоким и низким процентом невозврата кредита достаточно невелико. Существует вероятность перехода статуса потенциальных заемщиков из одной группы в другую. Поэтому данное предположение возможно учитывать в скоринговой модели банка, однакое оно не должно быть определеяющим параметром.

Предположение 3 частично подтверждено: недостаточно данных по клиентам с уровнем дохода в следующих группах: до 30 тыс. рублей, от 30 до 50 тыс. рублей и от 1 млн. рублей и выше.

    7,06% - доля просроченных кредитов у заемщиков с уровнем дохода от 200 до 1000
    8,50% - доля просроченных кредитов у заемщиков с уровнем дохода от 50 до 200

Рекомендация: провести дополнительное исследование на большей выборке данных.

Предположение 4 подтверждено:

    7,26% - доля просроченных кредитов, целью которых было приобретение недвижимости.
    7,91% - доля просроченных кредитов, целью которых было проведение свадьбы.
    9,35% - доля просроченных кредитов, целью которых было приобретение автомобиля.
    9,25% - доля просроченных кредитов, целью которых было получение образования.

Рекомендация: так как предположение подтверждено, данный параметр может быть включен в расчет скоринговой системы банка. Как и другие параметры банк должен рассматривать их в совокупности, оценивая общий шанс возврата кредита в срок.

Общие рекомендации:

1. необходимо учитывать параметры в совокупности.
2. специалистам банка можно учитывать зависимости из второго и четвертого предположения (семейное положение и цели кредита).
3. необходимо собрать более полную выборку для повторной проверки первого и третьего предположений.

